hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -1239,16 +1234,60 @@ public String getBasePath() {
   }
 
   public HoodieFileFormat getBaseFileFormat() {
-    return HoodieFileFormat.valueOf(getStringOrDefault(BASE_FILE_FORMAT));
+    return HoodieFileFormat.valueOf(getStringOrDefault(BASE_FILE_FORMAT).toUpperCase());
+  }
+
+  public String getRecordMergerStrategy() {
+    return getString(RECORD_MERGER_STRATEGY);
+  }
+
+  public RecordMergeMode getRecordMergeMode() {
+    return RecordMergeMode.valueOf(getString(RECORD_MERGE_MODE));
   }
 
   public HoodieRecordMerger getRecordMerger() {
-    List<String> mergers = StringUtils.split(getStringOrDefault(RECORD_MERGER_IMPLS), ",").stream()
+    List<String> mergers = StringUtils.split(getString(RECORD_MERGER_IMPLS), ",").stream()
         .map(String::trim)
         .distinct()
         .collect(Collectors.toList());
-    String recordMergerStrategy = getString(RECORD_MERGER_STRATEGY);
-    return HoodieRecordUtils.createRecordMerger(getString(BASE_PATH), engineType, mergers, recordMergerStrategy);
+    return getRecordMerger(getString(BASE_PATH), getRecordMergeMode(),
+        engineType, getLogDataBlockFormat(), mergers, getStringOpt(RECORD_MERGER_STRATEGY));
+  }
+
+  public static HoodieRecordMerger getRecordMerger(String basePath,
+                                                   RecordMergeMode mergeMode,
+                                                   EngineType engineType,
+                                                   HoodieLogBlock.HoodieLogBlockType logBlockType,
+                                                   List<String> mergers,
+                                                   Option<String> strategy) {
+    switch (logBlockType) {
+      case AVRO_DATA_BLOCK:
```

Comment:
For Avro data block, we should also use Spark merger and row-based merging unless `CUSTOM` merge mode is specified.  Only MDT should leverage Avro record merger.

Reply:
Also we should not couple log block type with merger type any more to simplify the supported merging logic.

Reply:
https://issues.apache.org/jira/browse/HUDI-5549 we do not have spark format support for avro log block

Reply:
Here is a ticket with subtasks of the issues I have found so far for writing with spark records https://issues.apache.org/jira/browse/HUDI-8317

Reply:
Sg.  Based on our previous discussion, we'll maintain the existing merge behavior as much as possible in this PR and tackle the adoption of Spark merger and row-based merging everywhere in the follow-ups.